packages/bitcore-node/src/modules/ripple/api/csp.ts

```diff
@@ -0,0 +1,140 @@
+      if (RippleStateProvider.client) {
+        await RippleStateProvider.client.getLedger();
+      }
+    } catch (e) {
+      RippleStateProvider.client = undefined;
+    }
+    if (!RippleStateProvider.client) {
+      const networkConfig = this.config[network];
+      const provider = networkConfig.provider;
+      const host = provider.host || 'localhost';
+      const protocol = provider.protocol || 'wss';
+      const portString = provider.port;
+      const connUrl = portString ? `${protocol}://${host}:${portString}` : `${protocol}://${host}`;
+      RippleStateProvider.client = new RippleAPI({ server: connUrl });
+      await RippleStateProvider.client.connect();
+    }
+    return RippleStateProvider.client;
+  }
+
+  async getBalanceForAddress(params: CSP.GetBalanceForAddressParams) {
+    const client = await this.getClient(params.network);
+    const info = await client.getAccountInfo(params.address);
+    const confirmed = Number(info.xrpBalance);
+    const balance = confirmed;
+    const unconfirmed = 0;
+    return { confirmed, unconfirmed, balance };
+  }
+
+  async getBlock(params: CSP.GetBlockParams) {
+    const client = await this.getClient(params.network);
+    const ledger = await client.getLedger({ includeTransactions: true, ledgerHash: params.blockId });
+    return ledger as any;
+  }
+
+  async getFee(params: CSP.GetEstimateSmartFeeParams) {
+    const client = await this.getClient(params.network);
+    const fee = await client.getFee();
+    return fee;
+  }
+
+  async broadcastTransaction(params: CSP.BroadcastTransactionParams) {
+    const client = await this.getClient(params.network);
+    return client.submit(params.rawTx);
+  }
+
+  async getWalletBalance(params: CSP.GetWalletBalanceParams) {
+    const { chain, network } = params;
+    const addresses = await this.getWalletAddresses(params.wallet._id!);
+    const balances = await Promise.all(
+      addresses.map(a => this.getBalanceForAddress({ address: a.address, chain, network, args: {} }))
+    );
+    return balances.reduce(
+      (total, current) => {
+        total.balance += current.balance;
+        total.confirmed += current.confirmed;
+        total.unconfirmed += current.unconfirmed;
+        return total;
+      },
+      { confirmed: 0, unconfirmed: 0, balance: 0 }
+    );
+  }
+
+  streamTxs<T>(txs: Array<T>, stream: Readable) {
+    for (let tx of txs) {
+      stream.push(tx);
+    }
+  }
+
+  async streamAddressTransactions(params: CSP.StreamAddressUtxosParams) {
+    const client = await this.getClient(params.network);
+    const txs = await client.getTransactions(params.address, {
+      start: params.args.startTx,
+      limit: Number(params.args.limit) || 100,
+      binary: false
+    });
+    const readable = new Readable({ objectMode: true });
+    this.streamTxs(txs, readable);
+    readable.push(null);
+    Storage.stream(readable, params.req, params.res);
+  }
+
+  async streamTransactions(params: CSP.StreamTransactionsParams) {
```

Comment:
This is for getting the TX from a certain block?

Reply:
Yeah, this should give you the transactions back for a given ledger, this method is likely not it's final implementation